src/portable/raspberrypi/rp2040/dcd_rp2040.c

```diff
@@ -103,14 +102,28 @@ static void _hw_endpoint_close(struct hw_endpoint *ep)
     *ep->buffer_control = 0;
     // Clear any endpoint state
     memset(ep, 0, sizeof(struct hw_endpoint));
+
+    // Reclaim buffer space if all endpoints are closed
+    bool reclaim_buffers = true;
+    for ( uint8_t i = 1; i < USB_MAX_ENDPOINTS; i++ )
+    {
+        if (hw_endpoint_get_by_num(i, TUSB_DIR_OUT)->hw_data_buf != NULL || hw_endpoint_get_by_num(i, TUSB_DIR_IN)->hw_data_buf != NULL)
+        {
+            reclaim_buffers = false;
+            break;
+        }
+    }
+    if (reclaim_buffers)
+    {
+        next_buffer_ptr = &usb_dpram->epx_data[0];
+    }
```

Comment:
I have an question, We only reclaim buffer when all endpoints are closed, this does help with audio only class. But the issue still exists if we are using composite with additional endpoints right ?

Reply:
That's true. A better solution would be to track and recycle individual chunks of memory. Or simply not deallocate when endpoints are closed.

Reply:
thank you for confirmation, it is still better than current state. It could buy me some time since I don't know when I could start refactoring.